org.OpenT2T.Sample.SuperPopular.Helpers/js/index.js

```diff
@@ -0,0 +1,167 @@
+'use strict';
+
+var https = require('https');
+var q = require('q');
+
+//this is our base, we refactor these options in each method
+ var options = 
+ {
+        protocol: 'https:',
+        host: 'api.wink.com'     
+ };
+    
+var  accessToken;  //we will need this when creating the headers
+      
+module.exports = 
+{
+    //get the initial parameters to build our target endpoint
+   initWinkApi: function(apiEndpoint, deviceID, token)
+   {
+       options.path = '/' + apiEndpoint + '/' + deviceID;
+       accessToken = token; //the specific token
+        
+   },
+ 
+   sendDesiredStateCommand: function(apiField,value) 
+   {
+       
+    //build our desired state object where the apiField is the target value
+    options.postData = {'desired_state':{}}; //we will add the field in this object
+    options.postData.desired_state[apiField] = value;
+    
+    var deferred = q.defer();   // q will help us with returning a promise
+    
+    //the headers to make our call
+    options.headers = 
+    {
+        'Authorization': 'Bearer ' + accessToken,
+        'Content-Type': 'application/json',
+        'Accept': 'application/json',
+        'Content-length':  JSON.stringify(options.postData).length
+    }
+    
+    options.method = 'PUT'; //the type of request to make
+   
+    //our HTTPS call 
+    var req = https.request(options, (res) => {
+        res.setEncoding('utf8');
+        res.on('data', (chunk) => 
```

Comment:
I would keep it there, it's consistent with all of the other HTTPS requests, and there maybe some calls in which we want to log from it.